snow/consensus/snowball/consensus.go

```diff
@@ -40,15 +40,19 @@ type Consensus interface {
 	Finalized() bool
 }
 
-// NnarySnowball augments NnarySnowflake with a counter that tracks the total
-// number of positive responses from a network sample.
-type NnarySnowball interface{ NnarySnowflake }
-
-// NnarySnowflake is a snowflake instance deciding between an unbounded number
-// of values. After performing a network sample of k nodes, if you have alpha
-// votes for one of the choices, you should vote for that choice. Otherwise, you
-// should reset.
-type NnarySnowflake interface {
+// ConsensusFactory produces NnarySnowflake instances
+type ConsensusFactory interface {
+	New(params Parameters, choice ids.ID) NnarySnow
```

Comment:
Could we be precise with both functions?
```suggestion
	NewNnary(params Parameters, choice ids.ID) NnarySnow
```